common/cache/domainCache.go

```diff
@@ -952,3 +958,15 @@ func (entry *DomainCacheEntry) IsSampledForLongerRetention(
 	}
 	return false
 }
+
+// MinutesToAddToRetentionDuration return number of hours to add to jittered wf deletion time
+func (entry *DomainCacheEntry) MinutesToAddToRetentionDuration() time.Duration {
+	if strMinutes, ok := entry.info.Data[JitteredWorkflowDeletionMinutesKey]; ok {
+		intVar, err := strconv.Atoi(strMinutes)
+		if err != nil {
+			return 0
+		}
+		return time.Minute * time.Duration(rand.Intn(intVar))
```

Comment:
Do you have a seed value specified for this btw? This will be deterministic if not, and you'll not end up with any jitter

Reply:
There is a rand.seed call inside host/service.go which is called in the service.go of each of the 4 actual services. The e2e tests I ran always had reasonably random numbers so I don't think it's deterministic.